docs/source/model.rst

```diff
@@ -1,14 +1,11 @@
 .. role:: hidden
     :class: hidden-section
 
-poutyne.framework
+poutyne.framework.Model
```

Comment:
Model is fine, it's the title of the page.

Reply:
I think if we want to be coherent, we should put a lowercase.

Reply:
This is for the tab on the left when opening the documentation. Would probably only make you think there is more than one class under the .model tab if it was lowercase.

Reply:
After discussion with @davebulaval, we think that the whole left section of the documentation page should be reviewed to be more intuitive than a simple package architecture (i.e. instead of 'poutyne.framework.callbacks', name the doc section 'Callbacks'). 

For now, let's leave this as is in this current PR, will do a PR soon for this.

Reply:
Yes I agree.